Cannot build on aarch64

I'm unable to build the latest sources (dev branch) on Arch Linux ARM (aarch64). Or to be more precise it's the linking that fails.
```
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen.c.o: in function `codegen_check_seg_read':
codegen.c:(.text+0x5824): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: codegen.c:(.text+0x582c): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen.c.o: in function `codegen_check_seg_write':
codegen.c:(.text+0x5a64): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: codegen.c:(.text+0x5a6c): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_block.c.o: in function `codegen_init':
codegen_block.c:(.text+0x98c): undefined reference to `codegen_backend_init'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ir.c.o: in function `codegen_ir_compile':
codegen_ir.c:(.text+0x9c): undefined reference to `codegen_backend_prologue'
/usr/bin/ld: codegen_ir.c:(.text+0xac): undefined reference to `uop_handlers'
/usr/bin/ld: codegen_ir.c:(.text+0xbc): undefined reference to `uop_handlers'
/usr/bin/ld: codegen_ir.c:(.text+0x168): undefined reference to `codegen_set_jump_dest'
/usr/bin/ld: codegen_ir.c:(.text+0x17c): undefined reference to `codegen_backend_epilogue'
/usr/bin/ld: codegen_ir.c:(.text+0x2b0): undefined reference to `codegen_set_jump_dest'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_branch.c.o: in function `ropJNP_common.constprop.0':
codegen_ops_branch.c:(.text+0x1354): undefined reference to `codegen_exit_rout'
/usr/bin/ld: codegen_ops_branch.c:(.text+0x1368): undefined reference to `codegen_exit_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_branch.c.o: in function `ropJP_common.constprop.0':
codegen_ops_branch.c:(.text+0x1850): undefined reference to `codegen_exit_rout'
/usr/bin/ld: codegen_ops_branch.c:(.text+0x1864): undefined reference to `codegen_exit_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_branch.c.o: in function `ropJO_common.constprop.0':
codegen_ops_branch.c:(.text+0x1da0): undefined reference to `codegen_exit_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_branch.c.o:codegen_ops_branch.c:(.text+0x1db8): more undefined references to `codegen_exit_rout' follow
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_fpu_loadstore.c.o: in function `ropFSTd':
codegen_ops_fpu_loadstore.c:(.text+0x2cac): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: codegen_ops_fpu_loadstore.c:(.text+0x2cb4): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: codegen_ops_fpu_loadstore.c:(.text+0x2ff0): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: codegen_ops_fpu_loadstore.c:(.text+0x2ff8): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_fpu_loadstore.c.o: in function `ropFSTPd':
codegen_ops_fpu_loadstore.c:(.text+0x39b4): undefined reference to `codegen_gpf_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_fpu_loadstore.c.o:codegen_ops_fpu_loadstore.c:(.text+0x39bc): more undefined references to `codegen_gpf_rout' follow
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_shift.c.o: in function `ropD2':
codegen_ops_shift.c:(.text+0x8b9c): undefined reference to `codegen_exit_rout'
/usr/bin/ld: codegen_ops_shift.c:(.text+0x8bb4): undefined reference to `codegen_exit_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_shift.c.o: in function `ropD3_w':
codegen_ops_shift.c:(.text+0x9a9c): undefined reference to `codegen_exit_rout'
/usr/bin/ld: codegen_ops_shift.c:(.text+0x9ab4): undefined reference to `codegen_exit_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_shift.c.o: in function `ropD3_l':
codegen_ops_shift.c:(.text+0xaa0c): undefined reference to `codegen_exit_rout'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_ops_shift.c.o:codegen_ops_shift.c:(.text+0xaa24): more undefined references to `codegen_exit_rout' follow
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_reg.c.o: in function `codegen_reg_writeback':
codegen_reg.c:(.text+0xd8): undefined reference to `codegen_direct_write_64'
/usr/bin/ld: codegen_reg.c:(.text+0x12c): undefined reference to `codegen_direct_write_64_stack'
/usr/bin/ld: codegen_reg.c:(.text+0x150): undefined reference to `codegen_direct_write_16'
/usr/bin/ld: codegen_reg.c:(.text+0x19c): undefined reference to `codegen_direct_write_8'
/usr/bin/ld: codegen_reg.c:(.text+0x1b4): undefined reference to `codegen_direct_write_8'
/usr/bin/ld: codegen_reg.c:(.text+0x1d4): undefined reference to `codegen_direct_write_double_stack'
/usr/bin/ld: codegen_reg.c:(.text+0x210): undefined reference to `codegen_direct_write_double'
/usr/bin/ld: codegen_reg.c:(.text+0x240): undefined reference to `codegen_direct_write_ptr'
/usr/bin/ld: codegen_reg.c:(.text+0x260): undefined reference to `codegen_direct_write_32_stack'
/usr/bin/ld: codegen_reg.c:(.text+0x26c): undefined reference to `codegen_direct_write_32'
/usr/bin/ld: codegen_reg.c:(.text+0x280): undefined reference to `codegen_direct_write_st_double'
/usr/bin/ld: codegen_reg.c:(.text+0x298): undefined reference to `codegen_direct_write_st_8'
/usr/bin/ld: codegen_reg.c:(.text+0x2ac): undefined reference to `codegen_direct_write_st_64'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_reg.c.o: in function `codegen_reg_reset':
codegen_reg.c:(.text+0x42c): undefined reference to `codegen_host_reg_list'
/usr/bin/ld: codegen_reg.c:(.text+0x434): undefined reference to `codegen_host_fp_reg_list'
/usr/bin/ld: codegen_reg.c:(.text+0x440): undefined reference to `codegen_host_reg_list'
/usr/bin/ld: codegen_reg.c:(.text+0x450): undefined reference to `codegen_host_fp_reg_list'
/usr/bin/ld: src/CMakeFiles/pcem.dir/codegen/codegen_reg.c.o: in function `codegen_reg_alloc_read_reg':
codegen_reg.c:(.text+0xe0c): undefined reference to `codegen_direct_read_pointer_stack'
/usr/bin/ld: codegen_reg.c:(.text+0xe80): undefined reference to `codegen_direct_read_32_stack'
/usr/bin/ld: codegen_reg.c:(.text+0xed4): undefined reference to `codegen_direct_read_64'
/usr/bin/ld: codegen_reg.c:(.text+0xf20): undefined reference to `codegen_direct_read_8'
/usr/bin/ld: codegen_reg.c:(.text+0xffc): undefined reference to `codegen_direct_read_16_stack'
/usr/bin/ld: codegen_reg.c:(.text+0x101c): undefined reference to `codegen_direct_read_st_8'
/usr/bin/ld: codegen_reg.c:(.text+0x102c): undefined reference to `codegen_direct_read_32'
/usr/bin/ld: codegen_reg.c:(.text+0x1048): undefined reference to `codegen_direct_read_st_64'
/usr/bin/ld: codegen_reg.c:(.text+0x107c): undefined reference to `codegen_direct_read_double_stack'
/usr/bin/ld: codegen_reg.c:(.text+0x10b0): undefined reference to `codegen_direct_read_64_stack'
/usr/bin/ld: codegen_reg.c:(.text+0x10f4): undefined reference to `codegen_direct_read_double'
/usr/bin/ld: codegen_reg.c:(.text+0x115c): undefined reference to `codegen_direct_read_16'
/usr/bin/ld: codegen_reg.c:(.text+0x1178): undefined reference to `codegen_direct_read_st_double'
/usr/bin/ld: codegen_reg.c:(.text+0x1188): undefined reference to `codegen_direct_read_pointer'
/usr/bin/ld: src/CMakeFiles/pcem.dir/cpu/386_dynarec.c.o: in function `opFINIT':
386_dynarec.c:(.text+0x6674): undefined reference to `codegen_set_rounding_mode'
/usr/bin/ld: src/CMakeFiles/pcem.dir/cpu/386_dynarec.c.o: in function `FLDENV.isra.0':
386_dynarec.c:(.text+0x21cb4): undefined reference to `codegen_set_rounding_mode'
/usr/bin/ld: 386_dynarec.c:(.text+0x21d94): undefined reference to `codegen_set_rounding_mode'
/usr/bin/ld: src/CMakeFiles/pcem.dir/cpu/386_dynarec.c.o: in function `FSAVE.isra.0':
386_dynarec.c:(.text+0x2225c): undefined reference to `codegen_set_rounding_mode'
/usr/bin/ld: src/CMakeFiles/pcem.dir/cpu/386_dynarec.c.o: in function `FSTOR.isra.0':
386_dynarec.c:(.text+0x26414): undefined reference to `codegen_set_rounding_mode'
/usr/bin/ld: src/CMakeFiles/pcem.dir/cpu/386_dynarec.c.o:386_dynarec.c:(.text+0x265ac): more undefined references to `codegen_set_rounding_mode' follow
collect2: error: ld returned 1 exit status
distcc[24838] ERROR: compile (null) on localhost failed
ninja: build stopped: subcommand failed.

```

--- a/cmake/arch-detect.cmake
+++ b/cmake/arch-detect.cmake
@@ -35,6 +35,8 @@ set(archdetect_c_code "
     #else
         #error cmake_ARCH ppc
     #endif
+#elif defined(__aarch64__) || defined(_M_ARM64)
+    #error cmake_ARCH arm64
 #endif
 
 #error cmake_ARCH unknown

